primitives/container-chain-genesis-data/src/json.rs

```diff
@@ -0,0 +1,286 @@
+                properties_json.get("ss58Format")
+            );
+
+            None
+        })
+    {
+        properties.token_metadata.ss58_format = x;
+    }
+    if let Some(x) = properties_json
+        .get("tokenDecimals")
+        .and_then(|x: &serde_json::Value| u32::try_from(x.as_u64()?).ok()).or_else(|| {
+            log::warn!(
+                "Failed to read properties.tokenDecimals from container chain chain spec, using default value instead. Invalid value was: {:?}",
+                properties_json.get("tokenDecimals")
+            );
+
+            None
+        })
+    {
+        properties.token_metadata.token_decimals = x;
+    }
+    if let Some(x) = properties_json.get("tokenSymbol").and_then(|x| {
+        let xs = x.as_str()?;
+        let xv: Vec<u8> = xs.to_string().into();
+
+        xv.try_into().ok()
+    }).or_else(|| {
+        log::warn!(
+            "Failed to read properties.tokenSymbol from container chain chain spec, using default value instead. Invalid value was: {:?}",
+            properties_json.get("tokenSymbol")
+        );
+
+        None
+    }) {
+        properties.token_metadata.token_symbol = x;
+    }
+    if let Some(x) = properties_json.get("isEthereum").and_then(|x| {
+        x.as_bool()
+    }).or_else(|| {
+        log::warn!(
+            "Failed to read properties.isEthereum from container chain chain spec, using default value instead. Invalid value was: {:?}",
+            properties_json.get("isEthereum")
+        );
+
+        None
+    }) {
+        properties.is_ethereum = x;
+    }
+
+    properties
+}
+
+pub fn properties_to_map(
+    properties: &Properties,
+) -> Result<serde_json::Map<String, serde_json::Value>, String> {
+    // TODO: we can just derive Serialize for genesis_data.properties instead of this hack,
+    // just ensure that the field names match. And "tokenSymbol" must be a string, in the struct
+    // it is defined as a Vec<u8>.
+    let properties = vec![
+        (
+            "ss58Format",
+            serde_json::Value::from(properties.token_metadata.ss58_format),
+        ),
+        (
+            "tokenDecimals",
+            serde_json::Value::from(properties.token_metadata.token_decimals),
+        ),
+        (
+            "tokenSymbol",
+            serde_json::Value::from(
+                String::from_utf8(properties.token_metadata.token_symbol.to_vec())
+                    .map_err(|e| format!("tokenSymbol is not valid UTF8: {}", e))?,
+            ),
+        ),
+        (
+            "isEthereum",
+            serde_json::Value::from(properties.is_ethereum),
+        ),
+    ]
+    .into_iter()
+    .map(|(k, v)| (k.to_string(), v))
+    .collect();
+
+    Ok(properties)
+}
+
+#[cfg(test)]
+mod tests {
+    use sp_core::ConstU32;
+
+    use super::*;
+
+    fn expected_container_chain_genesis_data() -> ContainerChainGenesisData<ConstU32<255>> {
```

Comment:
```suggestion
    fn expected_container_chain_genesis_data() -> ContainerChainGenesisData {
```

Why is the CI green when this shouldn't compile?

Reply:
Feature-gated `json`. Since locally tanssi compiles it seems this feature is not used?

Reply:
Probably that feature is enabled for build but disabled when running tests...

Reply:
I checked the CI script and we do the same as in the tanssi repo, so we will encounter this issue there as well, as long as there is some feature which is not enabled by default, our unit tests will not run :confused: 

I guess the easiest way to notice that is to check the generated coverage file